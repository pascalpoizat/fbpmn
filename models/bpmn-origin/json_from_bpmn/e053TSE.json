{
    "isInterrupt": {},
    "catN": {
        "Process_1": "Process",
        "EndEvent_0gtkoek": "NoneEndEvent",
        "NSE": "NoneStartEvent",
        "Process_1fvo71o": "Process",
        "Task_0x899gn": "AbstractTask",
        "StartEvent_04szy8h": "TimerStartEvent",
        "T1": "AbstractTask",
        "EE1": "NoneEndEvent"
    },
    "catE": {
        "SequenceFlow_12856fs": "NormalSequenceFlow",
        "SF2": "NormalSequenceFlow",
        "SF1": "NormalSequenceFlow",
        "SequenceFlow_0ru7vyn": "NormalSequenceFlow"
    },
    "containN": {
        "Process_1": [
            "NSE",
            "T1",
            "EE1"
        ],
        "Process_1fvo71o": [
            "StartEvent_04szy8h",
            "Task_0x899gn",
            "EndEvent_0gtkoek"
        ]
    },
    "messageE": {},
    "timeInformation": {
        "StartEvent_04szy8h": {
            "timerDefinitionValue": null,
            "timerDefinitionType": null
        }
    },
    "targetE": {
        "SequenceFlow_12856fs": "EndEvent_0gtkoek",
        "SF2": "EE1",
        "SF1": "T1",
        "SequenceFlow_0ru7vyn": "Task_0x899gn"
    },
    "name": "e053TSE",
    "attached": {},
    "containE": {
        "Process_1": [
            "SF1",
            "SF2"
        ],
        "Process_1fvo71o": [
            "SequenceFlow_0ru7vyn",
            "SequenceFlow_12856fs"
        ]
    },
    "edges": [
        "SF1",
        "SF2",
        "SequenceFlow_0ru7vyn",
        "SequenceFlow_12856fs"
    ],
    "messages": [],
    "sourceE": {
        "SequenceFlow_12856fs": "Task_0x899gn",
        "SF2": "T1",
        "SF1": "NSE",
        "SequenceFlow_0ru7vyn": "StartEvent_04szy8h"
    },
    "nodes": [
        "Process_1",
        "Process_1fvo71o",
        "NSE",
        "T1",
        "EE1",
        "StartEvent_04szy8h",
        "Task_0x899gn",
        "EndEvent_0gtkoek"
    ],
    "nameN": {
        "Task_0x899gn": "T2",
        "T1": "T1"
    }
}