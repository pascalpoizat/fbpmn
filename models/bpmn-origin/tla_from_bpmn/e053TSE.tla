---------------- MODULE e053TSE ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Process_1" :> {  }
  @@ "Process_1fvo71o" :> {  }

ContainRel ==
  "Process_1" :> { "NSE", "T1", "EE1" }
  @@ "Process_1fvo71o" :> { "StartEvent_04szy8h", "Task_0x899gn", "EndEvent_0gtkoek" }

Node == { "Process_1", "Process_1fvo71o", "NSE", "T1", "EE1", "StartEvent_04szy8h", "Task_0x899gn", "EndEvent_0gtkoek" }

Edge == { "SF1", "SF2", "SequenceFlow_0ru7vyn", "SequenceFlow_12856fs" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "SF1" :> "NSE"
@@ "SF2" :> "T1"
@@ "SequenceFlow_0ru7vyn" :> "StartEvent_04szy8h"
@@ "SequenceFlow_12856fs" :> "Task_0x899gn"

target ==
   "SF1" :> "T1"
@@ "SF2" :> "EE1"
@@ "SequenceFlow_0ru7vyn" :> "Task_0x899gn"
@@ "SequenceFlow_12856fs" :> "EndEvent_0gtkoek"

CatN ==
   "Process_1" :> Process
@@ "Process_1fvo71o" :> Process
@@ "NSE" :> NoneStartEvent
@@ "T1" :> AbstractTask
@@ "EE1" :> NoneEndEvent
@@ "StartEvent_04szy8h" :> TimerStartEvent
@@ "Task_0x899gn" :> AbstractTask
@@ "EndEvent_0gtkoek" :> NoneEndEvent

CatE ==
   "SF1" :> NormalSeqFlow
@@ "SF2" :> NormalSeqFlow
@@ "SequenceFlow_0ru7vyn" :> NormalSeqFlow
@@ "SequenceFlow_12856fs" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

