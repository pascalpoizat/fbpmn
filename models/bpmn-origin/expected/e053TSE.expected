Working in /tmp/e053TSE.YCm8k with 1 worker(s)
[92mtransformation done[0m
<<"Processes=", 2, "Nodes=", 8, "Gateway=", 0, "SF=", 4, "MF=", 0>>
---------- Network01Bag ----------
[X] Prop01Type
     states=25 trans=41 depth=9
[X] Prop02Safe
     states=25 trans=41 depth=9
[X] Prop03Sound
     states=25 trans=41 depth=9
[X] Prop04MsgSound
     states=25 trans=41 depth=9
---------- Network02FifoPair ----------
[X] Prop01Type
     states=25 trans=41 depth=9
[X] Prop02Safe
     states=25 trans=41 depth=9
[X] Prop03Sound
     states=25 trans=41 depth=9
[X] Prop04MsgSound
     states=25 trans=41 depth=9
---------- Network03Causal ----------
[X] Prop01Type
     states=25 trans=41 depth=9
[X] Prop02Safe
     states=25 trans=41 depth=9
[X] Prop03Sound
     states=25 trans=41 depth=9
[X] Prop04MsgSound
     states=25 trans=41 depth=9
---------- Network04Inbox ----------
[X] Prop01Type
     states=25 trans=41 depth=9
[X] Prop02Safe
     states=25 trans=41 depth=9
[X] Prop03Sound
     states=25 trans=41 depth=9
[X] Prop04MsgSound
     states=25 trans=41 depth=9
---------- Network05Outbox ----------
[X] Prop01Type
     states=25 trans=41 depth=9
[X] Prop02Safe
     states=25 trans=41 depth=9
[X] Prop03Sound
     states=25 trans=41 depth=9
[X] Prop04MsgSound
     states=25 trans=41 depth=9
---------- Network06Fifo ----------
[X] Prop01Type
     states=25 trans=41 depth=9
[X] Prop02Safe
     states=25 trans=41 depth=9
[X] Prop03Sound
     states=25 trans=41 depth=9
[X] Prop04MsgSound
     states=25 trans=41 depth=9
---------- Network07RSC ----------
[X] Prop01Type
     states=25 trans=41 depth=9
[X] Prop02Safe
     states=25 trans=41 depth=9
[X] Prop03Sound
     states=25 trans=41 depth=9
[X] Prop04MsgSound
     states=25 trans=41 depth=9
done.
